### PR TITLE
Add osp_deploy and osp_tempest target

### PR DESCRIPTION
**Pull request by stuggi**
For full end to end test this introduce targets to
run the OSP deployment playbooks and a tempest run.
The tempest use the osclient CRD to create a pod using the
tempest container image connected to specified osnetworks.

Full OSP deployment with tempest test run would be
1) run full deployment up to OSP is installed
make openstack

2) run tempest
make osp_tempest

Depends on: https://github.com/openstack-k8s-operators/osp-director-operator/pull/289

### CODE REVIEW

File: ansible/playbook_generator.yaml [hunk 1]
```diff
@@ -46,3 +46,10 @@
     environment:
       <<: *oc_env
     when: playbook_generator_switch.stdout | length == 0
+
+  - name: wait for playbook generator to finish
+    shell: |
+      set -e
+      oc wait osplaybookgenerator default --for condition=Finished --timeout="{{ (default_timeout * 3)|int }}s"
```
Comment (abays) [inline ansible/playbook_generator.yaml:53]: I think this needs `-n openstack`

Comment (stuggi) [reply ansible/playbook_generator.yaml:53]: yes, its also missing in a couple of other places. will update when I was able to test

File: ansible/osp_tempest.yaml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        oc cp -n openstack openstackclient:/home/cloud-admin/.config/openstack/clouds.yaml {{ tempest_working_dir }}/clouds.yaml
+      environment:
+        <<: *oc_env
+
+    - name: create /var/lib/tempest/.config/openstack directory and copy clouds.yaml
+      shell: |
+        #!/bin/bash
+        set -e
+        oc rsh -n openstack tempest mkdir -p /var/lib/tempest/.config/openstack
+        oc cp -n openstack {{ tempest_working_dir }}/clouds.yaml tempest:/var/lib/tempest/.config/openstack/clouds.yaml
```
Comment (abays) [inline ansible/osp_tempest.yaml:91]: Just curious here: does `oc cp` set the ownership of the copied file(s) to the `runUID`/`runGUID` of the pod?

Comment (stuggi) [reply ansible/osp_tempest.yaml:91]: yes it does as thats the default user when you rsh into the pod:

~~~
sh-4.4$ ls -la /var/lib/tempest/
total 24
drwx------. 1 tempest tempest  144 Jun 29 09:54 .
drwxr-xr-x. 1 root    root      21 Jun 23 08:26 ..
-rw-r--r--. 1 tempest tempest   18 Apr 21 14:04 .bash_logout
-rw-r--r--. 1 tempest tempest  141 Apr 21 14:04 .bash_profile
-rw-r--r--. 1 tempest tempest  376 Apr 21 14:04 .bashrc
drwxr-xr-x. 3 tempest tempest   23 Jun 29 09:53 .config
drwxr-xr-x. 2 tempest tempest   28 Jun 29 09:54 .tempest
-rw-r--r--. 1 tempest tempest  414 Jun 29 09:53 tempest-deployer-input.conf
-rw-r--r--. 1 tempest tempest 2043 Jun 29 09:53 tempest_script.sh
drwxr-xr-x. 6 tempest tempest  127 Jun 29 09:55 tempest_workspace
-rw-r--r--. 1 tempest tempest   81 Jun 29 09:53 whitelist_file
~~~

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**